packages/contracts/contracts/minter-suite/Minters/one-off/ClaimMinter.sol

```diff
@@ -0,0 +1,452 @@
+    string public constant minterType = "ClaimMinter";
+
+    /// The project ID for the Art Blocks project.
+    uint256 public immutable projectId;
+
+    /// Core contract address for this minter.
+    address public immutable coreContractAddress;
+
+    /// Minter filter address this minter interacts with
+    address public immutable minterFilterAddress;
+
+    /// PMP contract address this minter interacts with
+    address public immutable pmpContractAddress;
+
+    /// Pseudorandom atomic contract address this minter interacts with
+    address public immutable pseudorandomAtomicContractAddress;
+
+    /// Minter filter this minter may interact with.
+    IMinterFilterV1 public immutable minterFilter;
+
+    /// PMP contract this minter may interact with.
+    IPMPV0 public immutable pmpContract;
+
+    /// Pseudorandom atomic contract this minter may interact with.
+    IPseudorandomAtomic public immutable pseudorandomAtomicContract;
+
+    // claimed tokens using bitmaps (bitmapIndex => bitmap)
+    mapping(uint256 => uint256) public claimedBitmaps;
+
+    // pre-minted tokens using bitmaps (bitmapIndex => bitmap)
+    mapping(uint256 => uint256) public preMintedBitmaps;
+
+    uint256 public timestampStart;
+
+    // price configuration
+    uint256 public basePriceInWei;
+    uint256 public priceIncrementInWei;
+
+    // default price increment in wei for each token
+    uint256 public constant DEFAULT_PRICE_INCREMENT_IN_WEI = 500000000000000;
+
+    /**
+     * @notice Initializes contract to be a Filtered Minter for
+     * `minterFilter` minter filter.
+     * @param minterFilter_ Minter filter for which this will be a
+     * filtered minter.
+     * @param coreContract_ The only core contract address this minter will
+     * interact with.
+     * @param pmpContract_ The PMP contract address this minter will
+     * interact with.
+     * @param pseudorandomAtomicContract_ The pseudorandom atomic contract address this minter will
+     * interact with.
+     * @param projectId_ The project ID for the Art Blocks project.
+     */
+    constructor(
+        address minterFilter_,
+        address coreContract_,
+        address pmpContract_,
+        address pseudorandomAtomicContract_,
+        uint256 projectId_
+    ) ReentrancyGuard() {
+        minterFilterAddress = minterFilter_;
+        minterFilter = IMinterFilterV1(minterFilter_);
+        coreContractAddress = coreContract_;
+        pmpContractAddress = pmpContract_;
+        pmpContract = IPMPV0(pmpContract_);
+        pseudorandomAtomicContractAddress = pseudorandomAtomicContract_;
+        pseudorandomAtomicContract = IPseudorandomAtomic(
+            pseudorandomAtomicContract_
+        );
+        projectId = projectId_;
+    }
+
+    /**
+     * @notice Returns all claimed bitmaps
+     * @return An array of all claimed bitmaps
+     */
+    function getAllClaimedBitmaps() external view returns (uint256[] memory) {
```

Comment:
nit: order of functions (solidity design guide)

Reply:
🙏  updated in https://github.com/ArtBlocks/artblocks-contracts/pull/1857/commits/5964d5db534b18268f440e99717194483b28e807